src/nvim/eval/typval.c

```diff
@@ -1544,6 +1554,16 @@ bool tv_dict_get_callback(dict_T *const d,
   return res;
 }
 
+/// Set a dictionary as the return value
+void tv_dict_set(typval_T *rettv, dict_T *d)
```

Comment:
@ZyX-I named another function `tv_list_alloc_ret`, so following that pattern perhaps this should be named `tv_dict_set_ret`. And the other `tv_list_set_ret`.

Reply:
These need to be in the same level 2 section as `tv_*_alloc_ret`, not here. Though not the same level 3 section, perhaps either copy or its own. And I would really suggest to move functions to header and make them inline instead.

Also note that functions are not in line with the rest of the code: first argument should have different name, pointers should be `const`, first argument should be NONNULL, there should be documentation for each of the arguments.